stablehlo/compatibility/dialect/VersionedStablehloAttrs.td

```diff
@@ -0,0 +1,174 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+   Copyright 2022 The VersionedHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef STABLEHLO_COMPATIBILITY_VERSIONEDHLO_ATTRS
+#define STABLEHLO_COMPATIBILITY_VERSIONEDHLO_ATTRS
+
+include "stablehlo/compatibility/dialect/VersionedStablehloBase.td"
+include "stablehlo/compatibility/dialect/VersionedStablehloEnums.td"
+
+include "mlir/IR/AttrTypeBase.td"
+
+//===----------------------------------------------------------------------===//
+// Attributes
+//===----------------------------------------------------------------------===//
+
+def VersionedHLO_Dim : ArrayRefParameter<"int64_t", "Dimension"> {
+  let parser = "mlir::hlo::parseIntArray($_parser)";
+  let printer = "mlir::hlo::printIntArray($_printer, $_self)";
+}
+
+def VersionedHLO_ScatterDimensionNumbers : AttrDef<VersionedHLO_Dialect, "ScatterDimensionNumbers"> {
+  let cppNamespace = "::mlir::stablehlo";
+  let mnemonic = "scatter";
+  let summary = "Attribute that models the dimension information for scatter";
+  let parameters = (ins
+      VersionedHLO_Dim:$updateWindowDims,
+      VersionedHLO_Dim:$insertedWindowDims,
+      VersionedHLO_Dim:$scatterDimsToOperandDims,
+      "int64_t":$indexVectorDim
+  );
+  let assemblyFormat = "`<` struct(params) `>`";
```

Comment:
Want to avoid custom assembly if necessary. VHLO isn't really supposed to be kept in textual form (since textual assembly isn't stable) and ideally VHLO should be treated as an implementation detail. 

My goal was to keep as much of the versioned dialect in ODS as possible for maintainability. However when an attr has a mnemonic, some form of custom assembly must be specified, and this was a minimal option.

I've added a task for "figure out what to do for custom assembly in VHLO". There is no harm in sharing assembly formats, since we offer no stability guarantees on textual assembly. But this feels like an area of "if we offer it, someone will rely on it" since textual assembly is _relatively_ stable.